src/main/scala/uk/gov/hmrc/play/bootstrap/config/AuditingConfigProvider.scala

```diff
@@ -59,10 +61,15 @@ class AuditingConfigProvider @Inject()(
                 .getOrElse(throw new Exception("Missing consumer configuration for auditing"))
             ),
             auditSource = appName,
-            auditExtraHeaders = c.getOptional[Boolean]("auditExtraHeaders")
+            auditSentHeaders = {
```

Comment:
If this was written:
```scala
auditSentHeaders =
  c.getOptional[Boolean]("auditSentHeaders")
    .orElse(
      c.getOptional[Boolean]("auditExtraHeaders")
        .map(Function.const(_) {
          logger.warn("deprecated config key auditing.auditExtraHeaders - use auditing.auditSentHeaders")
        }
    ).getOrElse(false)
```
the order of precedence would be easier to follow, and the default value would be at the end. Also it wouldn't check `auditExtraHeaders` if `auditSentHeaders` was explicitly set to false.